python-sdk/src/astro/sql/operators/dataframe.py

```diff
@@ -215,6 +225,48 @@ def _convert_column_capitalization_for_output(function_output, columns_names_cap
             )
         return function_output
 
+    def get_openlineage_facets(self, task_instance) -> OpenLineageFacets:  # skipcq: PYL-W0613
+        """
+        Collect the input, output, job and run facets for DataframeOperator
+        """
+        input_dataset: list[OpenlineageDataset] = []
```

Comment:
why input_dataset is always none?

Reply:
because conceptually a dataframe isnt a Dataset (and its mandatory to specify input dataset as per OL)

Reply:
then just a thought do we need to declare this here can't we just pass a `[ ]` list 
```
return OpenLineageFacets(
            inputs=[], outputs=output_dataset, run_facets=run_facets, job_facets=job_facets
        )
```
WDYT?

Reply:
This is done